corehq/apps/reports/standard/deployments.py

```diff
@@ -247,13 +273,24 @@ def process_rows(self, users, fmt_for_export=False):
                 build_version = last_build.get('build_version') or build_version
                 if last_build.get('app_id'):
                     app_name = self.get_app_name(last_build['app_id'])
-            rows.append([
+
+            row_data = [
                 user_display_string(user.get('username', ''),
                                     user.get('first_name', ''),
                                     user.get('last_name', '')),
                 _fmt_date(last_seen, fmt_for_export), _fmt_date(last_sync_date, fmt_for_export),
                 app_name or "---", build_version, commcare_version or '---'
-            ])
+            ]
+
+            if (toggles.ICDS_DASHBOARD_REPORT_FEATURES.enabled(self.request.couch_user.username) and
+                    toggles.DASHBOARD_ICDS_REPORT.enabled(self.domain)) and self.rendered_as in ['export']:
```

Comment:
This is repeated several times, could pull it out into a `include_location_data` method, or move it into `required_loc_columns` like I said above, then if `required_loc_columns == []`, it'd continue to operate as normal, without needing to keep checking.